src/Microsoft.ML.TimeSeries/ExtensionsCatalog.cs

```diff
@@ -150,6 +148,29 @@ public static class TimeSeriesCatalog
             int windowSize = 64, int backAddWindowSize = 5, int lookaheadWindowSize = 5, int averageingWindowSize = 3, int judgementWindowSize = 21, double threshold = 0.3)
             => new SrCnnAnomalyEstimator(CatalogUtils.GetEnvironment(catalog), outputColumnName, windowSize, backAddWindowSize, lookaheadWindowSize, averageingWindowSize, judgementWindowSize, threshold, inputColumnName);
 
+        /// <summary>
+        /// Create <see cref="SrCnnAnomalyEstimator"/>, which detects timeseries anomalies using SRCNN algorithm.
+        /// </summary>
+        /// <param name="catalog">The transform's catalog.</param>
+        /// <param name="input">...</param>
+        /// <param name="outputColumnName">Name of the column resulting from the transformation of <paramref name="inputColumnName"/>.
+        /// The column data is a vector of <see cref="System.Double"/>. The vector contains 3 elements: alert (1 means anomaly while 0 means normal), raw score, and magnitude of spectual residual.</param>
+        /// <param name="inputColumnName">Name of column to transform. The column data must be <see cref="System.Single"/>.</param>
+        /// <param name="threshold">The threshold to determine anomaly, score larger than the threshold is considered as anomaly. Should be in (0,1)</param>
+        /// <param name="batchSize">.Divide the input data into batches to fit SrCnn model. Must be -1 or a positive integer no less than 12. Default value is 1024.</param>
+        /// <param name="sensitivity">The sensitivity of boundaries. Must be in the interval (0, 100).</param>
+        /// <param name="detectMode">The detect mode of the SrCnn model.</param>
+        /// <example>
+        /// <format type="text/markdown">
+        /// <![CDATA[
+        /// [!code-csharp[DetectAnomalyBySrCnn](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/TimeSeries/DetectAnomalyBySrCnnBatchPrediction.cs)]
+        /// ]]>
+        /// </format>
+        /// </example>
+        public static IDataView BatchDetectAnomalyBySrCnn(this DataOperationsCatalog catalog, IDataView input, string outputColumnName, string inputColumnName,
+            double threshold = 0.3, int batchSize = 1024, double sensitivity = 99, SrCnnDetectMode detectMode = SrCnnDetectMode.AnomalyOnly)
```

Comment:
Shouldn't this be in the TimeSeries catalog?

Reply:
It was the suggestion from @yaeldekel to put it here

Reply:
@yaeldekel, Can you please elaborate? We already have a TimeSeries catalog. Shouldn't it belong there?

---
In reply to: [426065077](https://github.com/dotnet/machinelearning/pull/5135#discussion_r426065077) [](ancestors = 426065077)

Reply:
In previous iterations it was in the `TransformsCatalog` which is why I suggested moving it. `TimeSeriesCatalog` would also work.

---
In reply to: [426078156](https://github.com/dotnet/machinelearning/pull/5135#discussion_r426078156) [](ancestors = 426078156,426065077)

Reply:
I see that RCA used a newly defined AnomalyDetectionCatalog, maybe it would be better to align with it?

---
In reply to: [426439074](https://github.com/dotnet/machinelearning/pull/5135#discussion_r426439074) [](ancestors = 426439074,426078156,426065077)

Reply:
Yes, Please move it to the AnomalyDetectionCatalog

---
In reply to: [427140758](https://github.com/dotnet/machinelearning/pull/5135#discussion_r427140758) [](ancestors = 427140758,426439074,426078156,426065077)